### PR TITLE
Use catalina for some osx jobs

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

Right now we're only testing in CI that things work on macOS high sierra
but mojave and catalina have been out for a while and different osx versions
differ enough that it warrants testing the newest version works. This
commit switches the osx jobs in the to use catalina. Additionally, azure pipelines
is removing support for the high sierra environment which means we'll have
to migrate to a newer version anyway.

### Details and comments

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -434,3 +434,53 @@ stages:
           inputs:
             testResultsFiles: '**/test-*.xml'
             testRunTitle: 'Test results for macOS Python $(python.version)'
+    - job: 'MacOS_Catalina_Tests'
```
Comment (kdk) [inline azure-pipelines.yml]:
```suggestion
    - job: 'MacOS_Mojave_Tests'
```

**Comment by mtreinish** [comment]
I'm actually thinking it might make more sense to just bump the existing osx jobs to mojave since this worked and is not a brand new release. I think the issues I was looking at before when we first migrated to azure must resolved now.

**Comment by mtreinish** [comment]
It looks like this is failing because of: https://github.com/Qiskit/qiskit-terra/issues/2793 (which is why we ended up using high sierra in CI originally) There's not much we can do since the only consistent factor involved is osx 10.14 which we don't have any ability to influence and/or fix.

**Comment by nonhermitian** [comment]
Conda install the needed numpy and scipy will work. They use mkl

**Comment by mtreinish** [comment]
The priority for this just went up, Azure Pipelines is going to stop offering the high sierra images starting Feb. 3, 2020. Which means we'll have to come up with a solution for running tests reliably on 10.14. I guess I'll have to pivot things to use conda instead of straight cpython and pip.

**Comment by mtreinish** [comment]
It looks like 10.15 is segfaulting too: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=8099&view=logs&j=377f2c8d-8b6f-583f-e152-2ff025e2543a&t=28b3ff0d-9e76-5e94-89e9-eed7321a383e&l=4178 I guess I'll still have to switch it to conda with catalina

File: azure-pipelines.yml [hunk 2]
```diff
@@ -234,8 +234,17 @@ stages:
           inputs:
             versionSpec: '$(python.version)'
           displayName: 'Use Python $(python.version)'
+        - bash: echo "##vso[task.prependpath]$CONDA/bin"
+          displayName: Add conda to PATH
```
Comment (kdk) [inline azure-pipelines.yml]: This looks like it just echos the conda path. Maybe it doesn't need to be added to PATH?

**Comment by mtreinish** [comment]
It looks like this segfaulted again even with conda: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=8866&view=logs&j=ce9a6331-8618-5e5b-ad8c-d3ab6621690a&t=b587a450-7392-5f12-37e1-082303ec0efb&l=4056

File: azure-pipelines.yml (hunk 2 above)
Comment (mtreinish) [reply azure-pipelines.yml]: I just copied this from: https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/anaconda?view=azure-devops&tabs=macos

**Comment by mtreinish** [comment]
Thanks to @nonhermitian this is now working on all versions of python with or without conda. This is ready to go whenever now

**Pull request merged**